src/hooks/useEscapeKey.js

```diff
@@ -0,0 +1,39 @@
+import React from "react";
+
+
+/**
+ * Executes provided callback whenever the user presses the Escape key.
+ * @param callback {function():void} Function executed after observing "keyup" event
+ */
+export const useEscapeKey = (callback) => {
+
+    // assign callback to a ref so react doesn't clear the function on re-render
+    // if some other state changes.
+    const cb = React.useRef();
+
+    React.useEffect(() => {
+        cb.current = callback;
+    }, [callback])
+
+    React.useEffect(() => {
+        function handleKeyUp (ev) {
+            if(!cb.current) {
+                return;
+            }
+            // compat: Firefox 36 and earlier maps Escape key to `Esc`
+            if(escapeKeyPressed(ev)) {
+                cb.current()
+            }
+        }
+        document.addEventListener("keyup", handleKeyUp)
+        return () => (document.removeEventListener("keyup", handleKeyUp))
+    }, [])
+
+}
```

Comment:
We save the callback on a ref for stability. We're not listing the hook as a dependency in the useEffect that registers the listener, though. So it's not _strictly_ necessary. 

see: https://balavishnuvj.com/blog/using-callbacks-in-custom-hooks/